signer/tests/integration/transaction_coordinator.rs

```diff
@@ -4258,6 +4360,21 @@ async fn process_rejected_withdrawal(is_completed: bool, is_in_mempool: bool) {
 #[test_case(false; "deposit not completed")]
 #[tokio::test]
 async fn coordinator_skip_onchain_completed_deposits(deposit_completed: bool) {
+    // We use this loop for just one purpose:
+    // Restart the test if was chosen different coordinator from one we want.
+    // It is hacky and not so nice, but I didn't find a nice way to grant necessary signer
+    // a coordinator role.
+    // In terms of performance it is not a problem, because if we chose wrong coordinator we restart pretty fast, and,
+    // probability of chosing correct coordinator is 50%.
+    loop {
+        let need_break = coordinator_skip_onchain_completed_deposits_inner(deposit_completed).await;
+        if need_break {
+            break;
+        }
+    }
+}
+
+async fn coordinator_skip_onchain_completed_deposits_inner(deposit_completed: bool) -> bool {
```

Comment:
Instead of doing all of this, I think we can change the test so that the current signer set is just one signer (themselves) with the bootstrap signer set is identical to it. It's still unsatisfactory since it's a complex setup, but then we do not need to loop through the test multiple times.